schema/formats.go

```diff
@@ -46,22 +47,53 @@ func (f macFormatChecker) IsFormat(input string) bool {
 	return match
 }
 
+func matchYangIPv4AddressPattern(input string) bool {
+	res, err := regexp.MatchString(`^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])(%[\p{N}\p{L}]+)?`, input)
+
+	if err != nil {
+		return false
+	}
+	return res
+}
+
+func (f ipv4FormatChecker) IsFormat(input string) bool {
+	ip := net.ParseIP(input)
+	if !((ip != nil) && (ip.To4() != nil)) {
+		return false
+	}
+	return matchYangIPv4AddressPattern(input)
+}
+
 func (f cidrFormatChecker) IsFormat(input string) bool {
-	_, _, err := net.ParseCIDR(input)
-	return err == nil
+	ip, _, err := net.ParseCIDR(input)
+	if err != nil {
+		return false
+	}
+	if ip.To4() != nil {
+		return matchYangIPv4AddressPattern(input)
+	}
+	return true
 }
 
 func (f cidrOrIPv4FormatChecker) IsFormat(input string) bool {
 	cidrIP, _, cidrErr := net.ParseCIDR(input)
 	ip := net.ParseIP(input)
-	return (cidrErr == nil && cidrIP.To4() != nil) || (ip != nil && ip.To4() != nil)
+
```

Comment:
could we reuse other format checkers here? sth. like

```Go
func (f cidrOrIPv4FormatChecker) IsFormat(input string) bool {
    ipv4 := ipv4FormatChecker{}
    cidr := ipv4NetworkFormatChecker{}
    return ipv4.IsFormat(input) || cidr.IsFormat(input)
}
```

Reply:
checkers are reused now.